app/controllers/dashboard_controller.rb

```diff
@@ -4,12 +4,36 @@ class DashboardController < ApplicationController
   after_action :discourage_caching
 
   def show
-    @incomplete_achievements = current_user.achievements.not_in_state(:dropped, :complete).with_courses.order("created_at DESC")
-    @completed_achievements = current_user.achievements.in_state(:complete).with_courses.order("updated_at DESC")
+    @incomplete_achievements = incomplete_achievements
+    @completed_achievements = completed_achievements
+    @enrolled_certificates, @unenrolled_certificates = enrolments
+  end
+
```

Comment:
I would make all these following methods private by adding a `private` at this point, otherwise they could be called directly in the future (if someone adds a route), but these are routes, just internal data handlers